### PR TITLE
Include editor config to enforce consistent styles

### PR DESCRIPTION
**Pull request by danielbachhuber**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_style = tab
+
+[*.json]
+indent_style = space
+indent_size = 2
```
Comment (danielbachhuber) [inline .editorconfig]: @goldenapples I copied this from Unified Post Types but don't we want to enforce tabs?

Comment (goldenapples) [inline .editorconfig]: Your call. I think the idea is that WordPress coding style is tabs, while node.js stuff - like `package.json`, etc - is more commonly done with two spaces, and it's easier to adopt prevailing style.

fwiw, I copied this directly from WordPress core's .editorconfig file. I'm not sure what the rule about json files was intended to refer to there.

Comment (danielbachhuber) [inline .editorconfig]: > fwiw, I copied this directly from WordPress core's .editorconfig file.

Ah. The distinction makes sense to me now.

**Comment by danielbachhuber** [comment]
@mpatek #reviewmerge ?

**Pull request merged**